Mathlib/Topology/Compactness/Lindelof.lean

```diff
@@ -693,45 +692,201 @@ instance {X : ι → Type*} [Countable ι] [∀ i, TopologicalSpace (X i)] [∀
+  rw [Subtype.isHereditarilyLindelof_iff, image_univ, Subtype.range_coe]
+
+instance {X : ι → Type*} [Countable ι] [∀ i, TopologicalSpace (X i)] [∀ i, LindelofSpace (X i)] :
+    LindelofSpace (Σi, X i) where
+  isLindelof_univ := by
+    rw [Sigma.univ]
+    exact isLindelof_iUnion fun i => isLindelof_range continuous_sigmaMk
+
 /-- Type class for Hereditarily Lindelöf spaces. -/
 class HereditarilyLindelofSpace (X : Type*) [TopologicalSpace X] : Prop where
   /-- In a Hereditarily Lindelöf space, `Set.univ` is a Hereditarily Lindelöf set. -/
   isHereditarilyLindelof_univ : IsHereditarilyLindelof (univ : Set X)
 
-lemma IsHereditarilyLindelof.isLindelof_subset (hs : IsHereditarilyLindelof s) (ht : t ⊆ s) :
-    IsLindelof t := hs t ht
+export HereditarilyLindelofSpace (isHereditarilyLindelof_univ)
 
-lemma IsHereditarilyLindelof.isLindelof (hs : IsHereditarilyLindelof s) :
-    IsLindelof s := hs.isLindelof_subset Subset.rfl
+protected alias IsHereditarilyLindelof.univ := isHereditarilyLindelof_univ
+
+theorem isHereditarilyLindelof_univ_iff :
+    IsHereditarilyLindelof (univ : Set X) ↔ HereditarilyLindelofSpace X :=
+  ⟨fun h => ⟨h⟩, fun h => h.1⟩
+
+theorem isHereditarilyLindelof_range [HereditarilyLindelofSpace X] {f : X → Y} (hf : Continuous f) :
+    IsHereditarilyLindelof (range f) := by
+  rw [← image_univ]; exact .image .univ hf
+
+theorem isHereditarilyLindelof_iff_HereditarilyLindelofSpace :
+    IsHereditarilyLindelof s ↔ HereditarilyLindelofSpace s :=
+  isHereditarilyLindelof_iff_isHereditarilyLindelof_univ.trans isHereditarilyLindelof_univ_iff
+
+lemma IsHereditarilyLindelof.of_coe [HereditarilyLindelofSpace s] : IsHereditarilyLindelof s :=
+  isHereditarilyLindelof_iff_HereditarilyLindelofSpace.mpr ‹_›
+
+protected theorem Topology.IsEmbedding.HereditarilyLindelofSpace
+    [h : HereditarilyLindelofSpace Y] {f : X → Y}
+    (hf : IsEmbedding f) : HereditarilyLindelofSpace X where
+  isHereditarilyLindelof_univ := by
+    rw [hf.isInducing.isHereditarilyLindelof_iff]
+    exact .subset .univ (subset_univ (f '' univ))
+
+/-- The disjoint union of two Hereditarily Lindelöf spaces is Hereditarily Lindelöf. -/
+instance [HereditarilyLindelofSpace X] [HereditarilyLindelofSpace Y] :
+    HereditarilyLindelofSpace (X ⊕ Y) where
+  isHereditarilyLindelof_univ := by
+    rw [← range_inl_union_range_inr]
+    exact (isHereditarilyLindelof_range continuous_inl).union
+      (isHereditarilyLindelof_range continuous_inr)
 
 instance (priority := 100) HereditarilyLindelof.to_Lindelof [HereditarilyLindelofSpace X] :
     LindelofSpace X where
-  isLindelof_univ := HereditarilyLindelofSpace.isHereditarilyLindelof_univ.isLindelof
+  isLindelof_univ := isHereditarilyLindelof_univ.isLindelof
 
+@[deprecated "Use `isHereditarilyLindelof_univ` and `IsHereditarilyLindelof.isLindelof_subset`"
+  (since := "2025-04-19")]
 theorem HereditarilyLindelof_LindelofSets [HereditarilyLindelofSpace X] (s : Set X) :
```

Comment:
Does `IsLindelof.of_coe` cover it now? If yes, we can just deprecate in favor of `IsLindelof.of_coe`.